### PR TITLE
[shardformer] fix opt test hanging

### PR DESCRIPTION
**Pull request by flybird11111**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [s] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [s] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

fix opt test hanging.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/amp/naive_amp/mixed_precision_mixin/fp16.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ def check_overflow(self) -> bool:
         self.found_overflow.fill_(0.0)
         if self.check_local_overflow():
             self.found_overflow.fill_(1.0)
+        print(f"begin all_reduce {dist.get_rank()} {self.found_overflow}")
```
Comment (ver217) [inline colossalai/amp/naive_amp/mixed_precision_mixin/fp16.py:59]: Remove print

File: tests/test_shardformer/test_model/test_shard_bert.py [hunk 2]
```diff
@@ -179,4 +205,4 @@ def test_bert_3d():
 
 if __name__ == "__main__":
     test_bert()
-    test_bert_3d()
+    # test_bert_3d()
```
Comment (ver217) [inline tests/test_shardformer/test_model/test_shard_bert.py:208]: Why do you comment this line?

Comment (flybird11111) [reply tests/test_shardformer/test_model/test_shard_bert.py]: > Why do you comment this line?

updated.

File: colossalai/amp/naive_amp/mixed_precision_mixin/fp16.py (hunk 1 above)
Comment (flybird11111) [reply colossalai/amp/naive_amp/mixed_precision_mixin/fp16.py]: > Remove print

updated.

**Review by ver217 — APPROVED** [review]

**Pull request merged**